src/Microsoft.AspNetCore.Mvc.Core/Internal/FileResultExecutorBase.cs

```diff
@@ -79,11 +79,10 @@ internal enum PreconditionState
             if (HttpMethods.IsHead(request.Method) || HttpMethods.IsGet(request.Method))
             {
                 var preconditionState = GetPreconditionState(context, httpRequestHeaders, lastModified, etag);
-                if (request.Headers.ContainsKey(HeaderNames.Range) &&
-                    (preconditionState == PreconditionState.Unspecified ||
+                if ((preconditionState == PreconditionState.Unspecified ||
```

Comment:
I have refactored it to check if the IfRange header satisfies the conditions before calling SetRangeHeader